File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s=="":
            return True
        if t=="":
            return False
        i = j = 0
        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                i+=1
            j+=1
            
            if i == len(s): return True

        return False

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int indexOfS = 0, indexOfT = 0;
        while(indexOfS < s.length() && indexOfT < t.length()) {
            if(s.charAt(indexOfS) == t.charAt(indexOfT)) {
                indexOfS++;
            }
            indexOfT++;
        }
        if(indexOfS == s.length()) return true;
        return false;
    }
}